soc/arm/nxp_lpc/lpc55xxx/soc.c

```diff
@@ -261,28 +260,20 @@ DT_FOREACH_STATUS_OKAY(nxp_lpc_ctimer, CTIMER_CLOCK_SETUP)
 #if defined(CONFIG_SOC_LPC55S36)
 	CLOCK_SetClkDiv(kCLOCK_DivAdc0Clk, 2U, true);
 	CLOCK_AttachClk(kFRO_HF_to_ADC0);
-
-#if defined(CONFIG_ADC_MCUX_LPADC)
-	/* Vref is required for LPADC reference */
-	POWER_DisablePD(kPDRUNCFG_PD_VREF);
-
-	vref_config_t vrefConfig;
-
-	VREF_GetDefaultConfig(&vrefConfig);
-	vrefConfig.bufferMode                     = kVREF_ModeHighPowerBuffer;
-	vrefConfig.enableInternalVoltageRegulator = true;
-	vrefConfig.enableVrefOut                  = true;
-	VREF_Init((VREF_Type *)VREF_BASE, &vrefConfig);
-#endif
-#else
+#else /* not LPC55s36 */
 	CLOCK_SetClkDiv(kCLOCK_DivAdcAsyncClk,
 			DT_PROP(DT_NODELABEL(adc0), clk_divider), true);
 	CLOCK_AttachClk(MUX_A(CM_ADCASYNCCLKSEL, DT_PROP(DT_NODELABEL(adc0), clk_source)));
 
 	/* Power up the ADC */
 	POWER_DisablePD(kPDRUNCFG_PD_LDOGPADC);
-#endif
-#endif
+#endif /* platform specific code */
+#endif /* DT has adc0 */
+
+#if (DT_NODE_HAS_COMPAT_STATUS(DT_NODELABEL(vref0), nxp_vref, okay))
+	CLOCK_EnableClock(kCLOCK_Vref);
+	POWER_DisablePD(kPDRUNCFG_PD_VREF);
+#endif /* vref0 */
```

Comment:
this needs to be handled by the regulator driver itself

Reply:
I agree but right now we are doing all of the clocking in soc.c, it would be hard to maintain if we have a different scheme for every peripheral, we are planning to redo the whole clocking scheme for nxp platforms at some point to do it correctly but for now I would like to remain consistent so that we can easily find the code that needs redone (which would all be in the soc level). Also, if I put this code in the regulator driver then there would need to be a bunch of #ifdef conditional compilation paths for every single different NXP device with this peripheral, which I would like to avoid

Reply:
it's "okay" then, clock control is a mess in Zephyr, this tiny change won't make things worse than they already are.